photutils/detection/core.py

```diff
@@ -250,19 +254,29 @@ def detect_sources(data, threshold, npixels, filter_kernel=None,
     return objlabels
 
 
-def find_peaks(data, threshold, min_separation=2, exclude_border=True,
-               segment_image=None, npeaks=np.inf, footprint=None):
+def find_peaks(data, threshold, box_size=3, footprint=None, mask=None,
+               border_width=None, npeaks=np.inf, subpixel=False, wcs=None):
     """
     Find local peaks in an image that are above above a specified
     threshold value.
 
-    Peaks are the local maxima above the ``threshold`` in a region of
-    ``(2 * min_separation) + 1`` (i.e., peaks are separated by at least
-    ``min_separation`` pixels).
+    Peaks are the maxima above the ``threshold`` within a local region.
+    The regions are defined by either the ``box_size`` or ``footprint``
+    parameters.  ``box_size`` defines the local region around each pixel
+    as a square box.  ``footprint`` is a boolean array where `True`
+    values specify the region shape.
 
-    If peaks are flat (i.e., multiple adjacent pixels have identical
-    intensities), then the coordinates of all such pixels are returned,
-    even if they are not separated by at least ``min_separation``.
+    If mulitple pixels within a local region have identical intensities,
+    then the coordinates of all such pixels are returned.  Otherwise,
+    there will be only one peak pixel per local region.  Thus, the
+    defined region effectively imposes a minimum separation between
+    peaks (unless there are identical peaks within the region).
+
+    When using subpixel precision (``subpixel=True``), then a cutout of
+    the specified ``box_size`` or ``footprint`` will be taken centered
+    on each peak and fit with a 2D Gaussian.  In this case, the fitted
```

Comment:
This sentence has to be updated with whatever exact method you choose: currently "subtract const, then fit 2d gauss".